hooks/useSubmission.tsx

```diff
@@ -408,130 +440,173 @@ function SubmissionControllerCreator({
+  );
+  submissionController.registerGenericDataType(
+    "submission_comments",
+    (item: unknown) => (item as SubmissionComments).id
+  );
+  submissionController.registerGenericDataType(
+    "submission_reviews",
+    (item: unknown) => (item as SubmissionReviewWithRubric).id
+  );
+  submissionController.registerGenericDataType(
+    "submission_artifact_comments",
+    (item: unknown) => (item as SubmissionArtifactComment).id
+  );
+
+  // Single comprehensive query to load all data upfront
+  const { query } = useShow<SubmissionWithAllRelatedData>({
     resource: "submissions",
     id: submission_id,
     meta: {
-      select:
-        "*, assignments(*, rubrics!grading_rubric_id(*,rubric_criteria(*,rubric_checks(*)))), submission_files(*), assignment_groups(*, assignment_groups_members(*, profiles!profile_id(*))), grader_results(*, grader_result_tests(*), grader_result_output(*)), submission_artifacts(*)"
+      select: `
+        *,
+        assignments(*, rubrics!grading_rubric_id(*,rubric_criteria(*,rubric_checks(*)))),
+        submission_files(*),
+        assignment_groups(*, assignment_groups_members(*, profiles!profile_id(*))),
+        grader_results(*, grader_result_tests(*), grader_result_output(*)),
+        submission_artifacts(*),
+        submission_file_comments(*),
+        submission_comments(*),
+        submission_reviews!submission_reviews_submission_id_fkey(*, rubrics(*, rubric_criteria(*, rubric_checks(*)))),
+        submission_artifact_comments!submission_artifact_comments_submission_id_fkey(*)
+      `.trim()
     }
   });
-  const { data: liveFileComments, isLoading: liveFileCommentsLoading } = useList<SubmissionFileComment>({
+
+  // Set up live subscriptions with proper event handling
+  // We need these enabled to receive live events, but we'll ignore the initial data since we already loaded it
+  const [liveSubscriptionsReady, setLiveSubscriptionsReady] = useState(false);
+
+  useList<SubmissionFileComment>({
     resource: "submission_file_comments",
     filters: [{ field: "submission_id", operator: "eq", value: submission_id }],
     liveMode: "manual",
-    pagination: {
-      pageSize: 1000
+    queryOptions: {
+      enabled: true, // Need to enable to receive live events
+      refetchOnMount: false, // Don't refetch on mount since we have data
+      refetchOnReconnect: false,
+      refetchOnWindowFocus: false
     },
     onLiveEvent: (event) => {
       submissionController.handleGenericDataEvent("submission_file_comments", event);
     }
   });
-  const { data: liveReviews, isLoading: liveReviewsLoading } = useList<SubmissionReviewWithRubric>({
+
+  useList<SubmissionReviewWithRubric>({
     resource: "submission_reviews",
     meta: {
       select: "*, rubrics(*, rubric_criteria(*, rubric_checks(*)))"
     },
     filters: [{ field: "submission_id", operator: "eq", value: submission_id }],
     liveMode: "manual",
-    pagination: {
-      pageSize: 1000
+    queryOptions: {
```

Comment:
What is the rationale for turning off refetchOnReconnect? IMO after reading the docs I think that it should be "always". With the goal of reducing refetches, however, I think that it should be possible to set the cache/stale time to infinity, as we will expect updates to occur as live events.